### PR TITLE
New users are not added to default group when using OIDCBackend

### PR DESCRIPTION
**Pull request by quadoss**
Jira: CDPD-20376

## What changes were proposed in this pull request?
Adding default group for the user when using OIDCBackend

### CODE REVIEW

File: desktop/core/src/desktop/auth/backend.py [hunk 1]
```diff
@@ -878,6 +878,7 @@ def create_user(self, claims):
         email=email,
         first_name=first_name,
         last_name=last_name,
+        groups=ensure_has_a_group(username),
```
Comment (wing2fly) [inline desktop/core/src/desktop/auth/backend.py:881]: function ensure_has_a_group is expecting user object not a username string. I think normal this function is calling from auth backend instead.

Comment (Harshg999) [reply desktop/core/src/desktop/auth/backend.py:881]: +1 for @wing2fly's comment. `ensure_has_a_group` needs a User object.

I think you can try like this:

```python
user = self.UserModel.objects.create_user(
    username=username,
    email=email,
    first_name=first_name,
    last_name=last_name,
    is_superuser=self.is_hue_superuser(claims)
)

ensure_has_a_group(user)
return user
```

Comment (quadoss) [reply desktop/core/src/desktop/auth/backend.py]: addressed it

**Review by Harshg999 — APPROVED** [review]
LGTM!

**Pull request merged**